File: comparativa/ejemplos/07-clases/04-redefine_instance_method.rb
class Dog
  def bark
    puts "Woof"
  end
end

foo = Dog.new()
foo.bark() # "Woof"

# replace bark instance method
class << foo
  def bark
    puts "Woof Woof"
  end
end

foo.bark() # "Woof Woof"

File: examples/2.high-level/python/11.poo/03.redefine/04-redefine_instance_method.py
class Dog:
    def bark(self):
        print "Woof"

def new_bark(self):
    print "Woof Woof"

foo = Dog()

funcType = type(Dog.bark)

# "Woof"
foo.bark()

# replace bark with new_bark for this object only
foo.bark = funcType(new_bark, foo, Dog)

foo.bark()
# "Woof Woof"
